### PR TITLE
Shadow: Make Event.composedPath() use closed-shadow-hidden when currentTarget is window

### LINKED ISSUE #326: event.composedPath() should not return every items in path if event.currentTarget is window
According to the current composedPath() definition, window.composedPath() would
return every items, even if the item's root is a closed shadow root, because
window is not a node.

### PR DESCRIPTION
**Pull request by hayatoito**
Fixes #326.

### CODE REVIEW

File: dom.bs [hunk 1]
```diff
@@ -744,11 +744,16 @@ steps:
...
-
-  <ol><li><p>If <var>currentTarget</var> is a <a>node</a> and <var>tuple</var>'s <b>item</b> is not
-  <a>closed-shadow-hidden</a> from <var>currentTarget</var>, or <var>currentTarget</var> is not a
-  <a>node</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.</p></li></ol>
-
+  <ol>
+   <li><p>If <var>currentTarget</var> is a {{Window}}, then:
+    <ol><li><p>If <var>tuple</var>'s <b>item</b> is not a <a>node</a>, or <var>tuple</var>'s
+    <b>item</b> is not <a>closed-shadow-hidden</a> from <var>tuple</var>'s <b>item</b>'s <a>node
+    document</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.</ol>
```
Comment (annevk) [inline dom.bs]: Why node document? Would shadow-including root not be a little cleaner if that's what you're going for? (I guess there's maybe a difference too, if the node is not currently in a document, although that would not yield a different result I think.)

File: dom.bs [hunk 2]
```diff
@@ -744,11 +744,16 @@ steps:
 
  <li>
   <p>For each <var>tuple</var> in <a>context object</a>'s <a for=Event>path</a>:
-
-  <ol><li><p>If <var>currentTarget</var> is a <a>node</a> and <var>tuple</var>'s <b>item</b> is not
-  <a>closed-shadow-hidden</a> from <var>currentTarget</var>, or <var>currentTarget</var> is not a
-  <a>node</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.</p></li></ol>
-
+  <ol>
+   <li><p>If <var>currentTarget</var> is a {{Window}}, then:
```
Comment (annevk) [inline dom.bs]: {{Window}} object

File: dom.bs [hunk 3]
```diff
@@ -744,11 +744,16 @@ steps:
 
  <li>
   <p>For each <var>tuple</var> in <a>context object</a>'s <a for=Event>path</a>:
-
-  <ol><li><p>If <var>currentTarget</var> is a <a>node</a> and <var>tuple</var>'s <b>item</b> is not
-  <a>closed-shadow-hidden</a> from <var>currentTarget</var>, or <var>currentTarget</var> is not a
-  <a>node</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.</p></li></ol>
-
+  <ol>
```
Comment (annevk) [inline dom.bs]: Newline before `<ol>`

File: dom.bs [hunk 4]
```diff
@@ -744,11 +744,16 @@ steps:
...
+   <li><p>If <var>currentTarget</var> is a {{Window}}, then:
+    <ol><li><p>If <var>tuple</var>'s <b>item</b> is not a <a>node</a>, or <var>tuple</var>'s
+    <b>item</b> is not <a>closed-shadow-hidden</a> from <var>tuple</var>'s <b>item</b>'s <a>node
+    document</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.</ol>
+
+   <li><p>Otherwise if <var>currentTarget</var> is a <a>node</a> and <var>tuple</var>'s <b>item</b>
+   is not <a>closed-shadow-hidden</a> from <var>currentTarget</var>, or <var>currentTarget</var> is
+   not a <a>node</a>, then append <var>tuple</var>'s <b>item</b> to <var>composedPath</var>.
+  </ol>
```
Comment (annevk) [inline dom.bs]: Newline after `</ol>`

File: dom.bs (hunk 1 above)
Comment (hayatoito) [inline dom.bs]: You are correct! I forgot that we can use `shadow-including root` here.

Comment (annevk) [inline dom.bs]: Also missing `</p></li>`

File: dom.bs (hunk 2 above)
Comment (annevk) [inline dom.bs]: Also needs a newline before the `<p>` since its not the only child of the `<li>`

Comment (hayatoito) [inline dom.bs]: Done.

File: dom.bs (hunk 1 above)
Comment (hayatoito) [inline dom.bs]: Done.

File: dom.bs (hunk 3 above)
Comment (hayatoito) [inline dom.bs]: Done.

**Comment by hayatoito** [comment]
Thank you for the review. I have updated the PR again.

File: dom.bs (hunk 4 above)
Comment (hayatoito) [inline dom.bs]: Done.

**Comment by annevk** [comment]
Just to be sure, the note after https://dom.spec.whatwg.org/#dom-event-composedpath still makes sense, right?

**Comment by hayatoito** [comment]
Yeah, I think this note remains true. Actually, I tried to update the note, but I found we do not have to update it.

**Comment by annevk** [comment]
So the only annoyance with merging this is that due to https://github.com/tabatkins/bikeshed/issues/815 we'll introduce a linking regression.

**Comment by annevk** [comment]
I guess I should just accept that for now.

**Pull request merged**